dashboard/pages/correlation/callbacks.py

```diff
@@ -146,7 +149,14 @@ def on_visualization_stage_entry(
     df = calculate_concentration(df_merged, concentration_value, volume_value)
 
     feature = "% ACTIVATION" if "% ACTIVATION_x" in df.columns else "% INHIBITION"
-    concentration_fig = concentration_plot(df, feature[2:])
+    concentration_fig = concentration_plot(
+        df[
+            df[f"{feature}_x"].between(
+                activity_threshold_bottom, activity_threshold_top
+            )
+        ],
```

Comment:
perhaps this could be put as a separate variable (imo would be slightly more readable)